src/wizards/tokenSwapDealWizard/stages/termsStage/termsStage.ts

```diff
@@ -87,4 +109,106 @@ export class TermsStage implements IBaseWizardStage {
     const isACreateWizard = [WizardType.createOpenProposal, WizardType.createPartneredDeal].includes(wizardType);
     return this.wizardState.registrationData.terms.clauses.map(() => isACreateWizard ? "edit" : "view");
   }
+
+  toggleDaoplomatRewards = (active: boolean) => {
+    if (active) {
+      this.daoplomatRewards = {
+        daoplomats: [],
+      };
+
+      this.wizardState.registrationData.terms.daoplomatRewards = this.daoplomatRewards;
+      this.addDaoplomatRewardsValidation();
+      this.addDaoplomat();
+    } else {
+      this.validationRules.off(this.daoplomatRewards);
+      this.daoplomatRewards = undefined;
+    }
+
+  };
+
+  addDaoplomat() {
+    const daoplomat: IDaoplomatReward = {
+      address: "",
+      rewardSplitPercentage: undefined,
+    };
+    this.daoplomatRewards?.daoplomats.push(daoplomat);
+    this.addDaoplomatValidation(daoplomat);
+  }
+
+  removeDaoplomat(index: number) {
+    this.daoplomatRewards.daoplomats.splice(index, 1);
+  }
+
+  private addDaoplomatValidation(daoplomat: IDaoplomatReward) {
+    this.validationRules
+      .on(daoplomat)
+      .ensure("address")
+      .required()
+      .withMessage("Please enter an address or an ENS name")
+      .satisfiesRule(new IsEthAddress())
+      .withMessage("Please enter a valid address or an ENS name")
+      .satisfies((value) => this.daoplomatRewards.daoplomats.filter(daoplomat => daoplomat.address === value).length === 1)
+      .withMessage("Address already used")
+      .ensure("rewardSplitPercentage")
+      .required()
+      .withMessage("Split is required")
+      .min(this.minimumSplitPercentage)
+      .withMessage("A reward distribution can not be lower than 0.001%")
+      .max(100)
+      .withMessage("The reward distribution should add up to 100%")
+    ;
+  }
+
+  private addDaoplomatRewardsValidation() {
+    if (!this.daoplomatRewards) {
+      return;
+    }
+    this.validationRules
+      .on(this.daoplomatRewards)
+      .ensureObject()
+      .satisfies((daoplomatRewards: IDaoplomatRewards) => {
+        return daoplomatRewards.daoplomats.reduce((total, daoplomat) => total + daoplomat.rewardSplitPercentage, 0) === 100;
+      })
+      .withMessage("The DAOplomat reward should add up to 100%.")
+      .ensure("percentage")
+      .required()
+      .withMessage("Please specify amount for the reward")
+      .min(this.minimumSplitPercentage)
+      .withMessage("The DAOplomat reward can not be lower than 0.001%")
+      .max(this.maximumSplitPercentage)
+      .withMessage("The DAOplomat reward can not be higher than 5%")
```

Comment:
Can we use the variables min and max here?
Or validation library does not allow interpolations?